support/cmake/JoinPaths.cmake

```diff
@@ -0,0 +1,23 @@
+# This module provides function for joining paths
+# known from from most languages
+#
+# SPDX-License-Identifier: (MIT OR CC0-1.0)
```

Comment:
Can this be distributed on {fmt}'s license (MIT with an optional exception for binary distribution): https://github.com/fmtlib/fmt/blob/master/LICENSE.rst? If yes I suggest replacing this with "# For the license information refer to LICENSE.rst."

Reply:
I would expect the CC0 (public domain like) license to be compatible with anything but I am not a lawyer. I can relicense it here under {fmt}’s license so we do not have to consider this.

Reply:
Ah, public domain is fine.